### PR TITLE
Add permissionless delegator

### PR DESCRIPTION
**Pull request by sukantoraymond**

### CODE REVIEW

**Review by dasconnor — APPROVED** [review]
LGTM, but please wait for felipe to review before merging

File: rpcpb/rpc.proto [hunk 1]
```diff
@@ -379,6 +386,14 @@ message AddPermissionlessValidatorResponse {
   ClusterInfo cluster_info = 1;
 }
 
+message AddPermissionlessDelegatorRequest {
+  repeated PermissionlessValidatorSpec validator_spec = 1;
```
Comment (felipemadero) [inline rpcpb/rpc.proto:390]: does it make sense to rename PermissioneslessValidatorSpec to PermissionlessStakerSpec?

File: cmd/control/control.go [hunk 2]
```diff
@@ -462,6 +473,34 @@ func transformElasticSubnetsFunc(_ *cobra.Command, args []string) error {
...
 }
 
+func addPermissionlessDelegatorFunc(_ *cobra.Command, args []string) error {
+	cli, err := newClient()
+	if err != nil {
+		return err
+	}
+	defer cli.Close()
+
+	validatorSpecStr := args[0]
```
Comment (felipemadero) [inline cmd/control/control.go:483]: should be validatorSpecsStr?

File: cmd/control/control.go [hunk 3]
```diff
@@ -462,6 +473,34 @@ func transformElasticSubnetsFunc(_ *cobra.Command, args []string) error {
...
+
+	validatorSpec := []*rpcpb.PermissionlessValidatorSpec{}
```
Comment (felipemadero) [inline cmd/control/control.go:485]: should be validatorSpecs?

File: local/blockchain.go [hunk 4]
```diff
@@ -175,6 +175,16 @@ func (ln *localNetwork) AddPermissionlessValidators(
 	return ln.addPermissionlessValidators(ctx, validatorSpec)
 }
 
+func (ln *localNetwork) AddPermissionlessDelegators(
+	ctx context.Context,
+	validatorSpec []network.PermissionlessValidatorSpec,
```
Comment (felipemadero) [inline local/blockchain.go:180]: should be delegatorSpecs?

File: local/blockchain.go [hunk 5]
```diff
@@ -897,6 +907,95 @@ func (ln *localNetwork) removeSubnetValidators(
 	return ln.restartNodes(ctx, nil, nil, nil, removeSubnetSpecs, nil)
 }
 
+func (ln *localNetwork) addPermissionlessDelegators(
+	ctx context.Context,
+	validatorSpecs []network.PermissionlessValidatorSpec,
```
Comment (felipemadero) [inline local/blockchain.go:912]: should be delegatorSpecs?

File: server/network.go [hunk 6]
```diff
@@ -293,6 +293,45 @@ func (lc *localNetwork) CreateChains(
 	return chainIDs, nil
 }
 
+func (lc *localNetwork) AddPermissionlessDelegators(ctx context.Context, validatorSpecs []network.PermissionlessValidatorSpec) error {
+	lc.lock.Lock()
+	defer lc.lock.Unlock()
+
+	if len(validatorSpecs) == 0 {
+		ux.Print(lc.log, logging.Orange.Wrap(logging.Bold.Wrap("no validator specs provided...")))
```
Comment (felipemadero) [inline server/network.go:301]: delegator specs provided?

File: server/network.go [hunk 7]
```diff
@@ -293,6 +293,45 @@ func (lc *localNetwork) CreateChains(
 	return chainIDs, nil
 }
 
+func (lc *localNetwork) AddPermissionlessDelegators(ctx context.Context, validatorSpecs []network.PermissionlessValidatorSpec) error {
```
Comment (felipemadero) [inline server/network.go:296]: should be delegatorSpecs?

File: tests/e2e/e2e_test.go [hunk 8]
```diff
@@ -1023,6 +1023,19 @@ var _ = ginkgo.Describe("[Start/Remove/Restart/Add/Stop]", func() {
 		})
 	})
 
+	ginkgo.It("add permissionless delegator to elastic subnets", func() {
+		ginkgo.By("adding a permissionless delegator to permissionless validator", func() {
+			ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+			defer cancel()
+			testValidatorConfig.SubnetId = createdSubnetID
+			testValidatorConfig.AssetId = elasticAssetID
+			testValidatorConfig.NodeName = "permissionlessNode"
```
Comment (felipemadero) [inline tests/e2e/e2e_test.go:1032]: Can we make this name a constant?

**Review by felipemadero — APPROVED** [review]

**Pull request merged**